pkg/client/unversioned/request.go

```diff
@@ -115,17 +115,22 @@ type Request struct {
 }
 
 // NewRequest creates a new request helper object for accessing runtime.Objects on a server.
-func NewRequest(client HTTPClient, verb string, baseURL *url.URL, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
+func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPath string, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
 	if backoff == nil {
 		glog.V(2).Infof("Not implementing request backoff strategy.")
 		backoff = &NoBackoff{}
 	}
 	metrics.Register()
+
+	pathPrefix := "/"
+	if baseURL != nil {
+		pathPrefix = path.Join(pathPrefix, baseURL.Path)
+	}
 	return &Request{
 		client:       client,
 		verb:         verb,
 		baseURL:      baseURL,
-		path:         baseURL.Path,
+		path:         path.Join(pathPrefix, versionedAPIPath),
```

Comment:
`path` is really a bad name for this variable. it's actually treated as a "path prefix" (see https://github.com/mesosphere/kubernetes/blob/jdef_abspath_breaks_proxies/pkg/client/unversioned/request.go#L143)

so I think we're actually doing the right thing here. i'm happy to rename this for clarity -- would that satisfy?